### ISSUE TITLE
Remove signature of SecretReveal

### ISSUE THREAD

**Issue by andrevmatos** [raiden-network/raiden#3513]
## Abstract

Currently, [`RevealSecret` is a `SignedMessage`](https://github.com/raiden-network/raiden/blob/5b67eb46a7c7bb12f29f5dccce958ee33ec4951b/raiden/messages.py#L722-L787), which requires a signature and [it's recovered and validated as soon as it gets to the transport](https://github.com/raiden-network/raiden/blob/5b67eb46a7c7bb12f29f5dccce958ee33ec4951b/raiden/network/transport/matrix/transport.py#L815-L829), therefore making the signature mandatory.
But, actually, as stated on https://github.com/raiden-network/raiden/issues/3254#issuecomment-464799513 , there's no need for this message to be a `SignedMessage`, as `secret` is enough for validating it:
- If `secret` is valid (i.e. it has the same hash as the `secrethash`), the message is valid, it doesn't matter if it somehow could be sent by someone else, as only the initiator knows the secret; 
- If `secret` is invalid, doesn't matter if it actually came from the initiator, it can't be used and is not useful

## Specification

We need to remove the requirement for this message to be signed, by allowing it to go through the transport even if it isn't signed, and only validate it on the state machine. The `sender` attribute can still be there, but filled by the transport instead of recovered from the signature.

- [ ] Remove requirement for certain classes of messages (for now, `RevealSecret`) to be signed to pass through transport, and, in these specific cases, set the `sender` explicitly from the recovered `peer_address` (for Matrix)

A lot of what is needed to implement it (`AuthenticatedMessage` supertype) is being done in #3504 . `RevealSecret` can become another type of `AuthenticatedMessage` which isn't `SignedMessage` and has its `sender` filled by transport.

## Backwards Compatibility

We can keep sending signed `RevealSecret` for now, for backwards compatibility. This issue is just about not **requiring** it, so clients that send unsigned versions of this message with valid secrets can still comply with new clients.

**Comment by hackaugusto** [comment]
> there's no need for this message to be a `SignedMessage`, as `secret` is enough for validating it:

No, no, the secret is not enough, we need to know who sent the `SecretReveal` message, IOW this message has to be authenticated, the means which authentication happens can be something other then a signature.

**Comment by hackaugusto** [comment]
> We can keep sending signed RevealSecret for now, for backwards compatibility. This issue is just about not requiring it, so clients that send unsigned versions of this message with valid secrets can still comply with new clients

To clarify, this is **not** backwards compatible. The change is *remove the signature*, if we do, this will break backwards compatibility. This either requires a breaking change or a protocol upgrade.

**Comment by andrevmatos** [comment]
> A lot of what is needed to implement it (`AuthenticatedMessage` supertype) is being done in #3504 . `RevealSecret` can become another type of `AuthenticatedMessage` which isn't `SignedMessage` and has its `sender` filled by transport.

> The `sender` attribute can still be there, but filled by the transport instead of recovered from the signature.

As I said, of course the sender would still be set, but this proposal is about not **requiring** it to be done through the overhead of a full signature, can be done e.g. on matrix by explicitly setting `sender` from the user signature (done once, not per message). Nonetheless, I don't see why we actually require the sender on the message, we could to the same effect verify all pending secrethashes and act if there's any match.

> To clarify, this is **not** backwards compatible. The change is *remove the signature*, if we do, this will break backwards compatibility. This either requires a breaking change or a protocol upgrade.

I don't agree with this statement. If you refer to the OP, it's about the **requirement** of the message on the receiving transport and state machine. We **can** (for now) continue sending the signed messages, as it is currently, making it backwards compatible, only on **receiving** we would not require the signature to be set, accepting either signed and unsigned messages and setting `sender` accordingly.
